src/libs/Localize/index.ts

```diff
@@ -82,39 +79,65 @@ const translationCache = new Map<ValueOf<typeof CONST.LOCALES>, Map<TranslationP
  * available in the given locale. If it is, it translates the
  * phrase and stores the translated value in the cache and returns
  * the translated value.
- *
- * @param language
- * @param phraseKey
- * @param fallbackLanguage
- * @param phraseParameters
  */
-function getTranslatedPhrase<TKey extends TranslationPaths>(
+function getTranslatedPhrase<TPath extends TranslationPaths>(
     language: 'en' | 'es' | 'es-ES',
-    phraseKey: TKey,
-    fallbackLanguage: 'en' | 'es' | null = null,
-    ...phraseParameters: PhraseParameters<Phrase<TKey>>
+    path: TPath,
+    fallbackLanguage: 'en' | 'es' | null,
+    ...parameters: TranslationParameters<TPath>
 ): string | null {
     // Get the cache for the above locale
     const cacheForLocale = translationCache.get(language);
 
     // Directly access and assign the translated value from the cache, instead of
     // going through map.has() and map.get() to avoid multiple lookups.
-    const valueFromCache = cacheForLocale?.get(phraseKey);
+    const valueFromCache = cacheForLocale?.get(path);
 
     // If the phrase is already translated, return the translated value
     if (valueFromCache) {
         return valueFromCache;
     }
 
-    const translatedPhrase = translations?.[language]?.[phraseKey] as Phrase<TKey>;
+    const translatedPhrase = translations?.[language]?.[path];
 
     if (translatedPhrase) {
         if (typeof translatedPhrase === 'function') {
-            return translatedPhrase(...phraseParameters);
+            /**
+             *
+             * is Plain object is for checking if the phraseTranslated output
+             * is an object then further check if it include the count param or not
+             * OR before checking the plain object output, we can check if we have the count
+             * param in parameters
+             *
+             */
+            const translateFunction = translatedPhrase as unknown as (...parameters: TranslationParameters<TPath>) => string | PluralForm;
+            const translateResult = translateFunction(...parameters);
+
+            if (typeof translateResult === 'string') {
+                return translateResult;
+            }
+
+            const phraseObject = parameters[0];
+            if (phraseObject && typeof phraseObject === 'object' && 'count' in phraseObject && typeof phraseObject.count === 'number') {
+                const pluralRule = new Intl.PluralRules(language).select(phraseObject.count);
+
+                const pluralResult = pluralRule in translateResult && translateResult[pluralRule];
+                if (pluralResult) {
+                    if (typeof pluralResult === 'string') {
+                        return pluralResult;
+                    }
+
+                    return pluralResult(phraseObject.count);
```

Comment:
I don't understand why is this problematic? 🤔

Reply:
I think it was a design choice when we come up with `PluralForm` type, however we might be able to move the count to the outer function, so this is supported:
```ts
  key2: ({count}) => ({
        one: '1 b',
        other: `${count} bs`,
    }),
```

Reply:
(I'm just helping with typescript side of things as this is quite complicated haha, design choice are yours @ZhenjaHorbach @shubham1206agra)

Reply:
> We do need to pass count as a parameter otherwise it will fail to derive the correct value.

That isn't true, check this example, you don't need to pass count parameter in param types:

```ts
    key2: () => ({
        one: '1 b',
        other: (count: number) => `${count} bs`,
    }),
```

Reply:
Why?

Reply:
@shubham1206agra I think you are missing on the changes introduced in types.ts file, check out new type for parameters:
```ts
/**
 * Determines the expected parameters for a specific translation function based on the provided translation path
 */
type TranslationParameters<TPath extends TranslationPaths> = FlatTranslationsObject[TPath] extends (...args: infer Args) => infer Return
    ? Return extends PluralForm
        ? Args[0] extends undefined
            ? [PluralParams]
            : [Args[0] & PluralParams]
        : Args
    : never[];
```